src/hooks/useFundSponsorship.ts

```diff
@@ -0,0 +1,47 @@
+import { useCallback } from 'react'
+import {
+    getTokenAndBalanceForSponsorship,
+    TokenAndBalanceForSponsorship,
+} from '~/getters/getTokenAndBalanceForSponsorship'
+import { useWalletAccount } from '~/shared/stores/wallet'
+import { toaster } from 'toasterhea'
+import FundSponsorshipModal from '~/modals/FundSponsorshipModal'
+import { Layer } from '~/utils/Layer'
+import { errorToast } from '~/utils/toast'
+import { fundSponsorship } from '~/services/sponsorships'
+
+const fundSponsorshipModal = toaster(FundSponsorshipModal, Layer.Modal)
+export const useFundSponsorship = (): ((
```

Comment:
We can make a `fundSponsorship` utility that takes the wallet address as an argument making is usable in react, or in services, or anywhere. No need for a hook.

Reply:
Here I'd prefer to stay with a hook. This hook allows us to trigger the whole modal and its process from multiple places without thinking of the implementation of this process and its required parameters. Currently it's being triggered from two different places. I don't want to put the UI logic (the modal etc) into a service that contains mostly infrastructure logic. React hooks are not bad :)